lib/ui/popup.js

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['Popup'];
+
+function Popup(opts) {
+  this.win = opts.win;
+  this.events = opts.events;
+
+  this.onPopupHiding = this.onPopupHiding.bind(this);
```

Comment:
@chuckharmston updated the callback to `beforePopupHide`, and the event name to `before-popup-hide`, thoughts?